pkg/operation/operation.go

```diff
@@ -164,6 +169,51 @@ func (o *Operation) InitializeShootClients() error {
 	return nil
 }
 
+// InitializeMonitoringClient will read the Prometheus ingress auth and tls
+// secrets from the Seed cluster, which are containing the cert to secure
+// the conntection and the credentials authenticate against the Shoot Prometheus.
+// With those certs and credentials, a Prometheus client API will be created
+// and attached to the existing Operation object.
+func (o *Operation) InitializeMonitoringClient() error {
+	if o.MonitoringClient != nil {
+		return nil
+	}
+
+	// Read the CA.
+	tlsSecret, err := o.K8sSeedClient.GetSecret(o.Shoot.SeedNamespace, "prometheus-tls")
+	if err != nil {
+		return err
+	}
+	if tlsSecret == nil {
```

Comment:
In general OK, but we never do this kind of check (the `if err != nil` check suffices).